src/sv-pipeline/scripts/validate_ped.py

```diff
@@ -0,0 +1,138 @@
+    with open(ped_file, 'r') as ped:
+        for line in ped:
+            # specification allows commented lines, which should be removed by SubsetPedFile task
+            if line.startswith("#"):
+                continue
+
+            # we require tab-delimited PED files although the specification does not
+            fields = line.strip().split("\t")
+            if len(fields) < 6:
+                raise ValueError("Invalid PED file. PED file must be tab-delimited and have 6 columns: " +
+                                 "family ID, sample ID, paternal ID, maternal ID, sex, and phenotype.")
+
+            # validate IDs
+            # don't check for duplicates here:
+            # family and parent IDs may appear multiple times, and sample IDs checked elsewhere
+            for identifier, id_type in zip(fields[:FIELD_NUMBER_SEX], ["family", "sample", "parent", "parent"]):
+                if id_type == "parent" and identifier == "0":
+                    continue  # allow maternal and paternal ID to be 0 (otherwise all-numeric IDs not allowed)
+                validate_id(identifier, None, id_type, "PED file")
+
+            # check for at least one appearance each of 1 and 2 in sex column
+            # this check is to ensure M/F are coded as 1 and 2 (not 0 and 1)
+            # and both M and F are present (for sex-specific steps)
+            sample_id = fields[FIELD_NUMBER_ID]
+            sex = fields[FIELD_NUMBER_SEX]
+            if sex == "1":
+                seen_sex_1 = True
+            elif sex == "2":
+                seen_sex_2 = True
+
+            # require integer sex
+            elif not sex.isdigit():
+                raise ValueError(f"Sample {sample_id} has an invalid value for sex: {sex}. " +
+                                 "PED file must use integer values for sex: 1=Male, 2=Female, 0=Unknown/Other.")
+
+            # check all samples in samples list are present in PED file exactly once
+            # no duplicate sample IDs
+            if sample_id in samples_found:
+                raise ValueError(f"Invalid PED file. PED file has duplicate entries for sample {sample_id}.")
+            elif sample_id in samples:
+                samples_found.add(sample_id)
+            # if not in samples list, ignore: ok for PED file to contain samples not in sample list
+
+    # check if all samples in sample list are present in PED file
+    if len(samples_found) < len(samples):
+        missing = samples - samples_found
+        raise ValueError(f"Invalid PED file. PED file is missing sample(s): {','.join(missing)}.")
+
+    if not (seen_sex_2 and seen_sex_1):
+        raise ValueError("Invalid PED file. PED file must use integer values for sex: " +
+                         "1=Male, 2=Female, 0=Unknown/Other. PED file must contain at least " +
+                         "one sample with sex=1 and one with sex=2.")
+
+    # passed validation tests
+    sys.stderr.write("PED file passes validation!\n")
```

Comment:
This seems like a success message, is writing to `stderr` intentional?

Reply:
Yes, it's a success message. I originally was writing to stdout, but thought it might make more sense for all the messages to be printed to one location. I'm happy to change it back if that makes more sense.

Reply:
Thanks for clarifying. I think `stdout` would make more sense in this case.